.ci/pitest-suppressions/pitest-regexp-suppressions.xml

```diff
@@ -0,0 +1,101 @@
+
+  <mutation unstable="false">
+    <sourceFile>MultilineDetector.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.MultilineDetector</mutatedClass>
+    <mutatedMethod>findMatch</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to java/util/regex/Pattern::toString</description>
+    <lineContent>options.getReporter().log(1, MSG_STACKOVERFLOW, matcher.pattern().toString());</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RegexpCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck</mutatedClass>
+    <mutatedMethod>beginTree</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
+    <description>Removed assignment to member variable errorCount</description>
+    <lineContent>errorCount = 0;</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RegexpCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck</mutatedClass>
+    <mutatedMethod>beginTree</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
+    <description>Removed assignment to member variable matchCount</description>
+    <lineContent>matchCount = 0;</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RegexpCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck</mutatedClass>
+    <mutatedMethod>isIgnore</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator</mutator>
+    <description>removed call to com/puppycrawl/tools/checkstyle/api/FileText::lineColumn</description>
+    <lineContent>end = text.lineColumn(0);</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RegexpCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpCheck</mutatedClass>
+    <mutatedMethod>setDuplicateLimit</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
+    <description>Removed assignment to member variable duplicateLimit</description>
+    <lineContent>this.duplicateLimit = duplicateLimit;</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>RegexpMultilineCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.regexp.RegexpMultilineCheck</mutatedClass>
+    <mutatedMethod>getRegexCompileFlags</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.ReturnValuesMutator</mutator>
+    <description>replaced return of primitive boolean/byte/short/integer value with (x == 1) ? 0 : x + 1</description>
+    <lineContent>return result;</lineContent>
+  </mutation>
```

Comment:
@Vyom-Yadav please open PR with hardcoded mutation for this one.

Reply:
#11999, I won't be adding comment explanation comment at https://github.com/checkstyle/checkstyle/issues/11719, this particular mutant is very well documented at https://pitest.org/quickstart/mutators/#RETURN_VALS